### ISSUE TITLE
Unable to set searchLogic for an Accessor column

### ISSUE THREAD

**Issue by kiddtang** [Laravel-Backpack/CRUD#1114]
### What I did:
```
$this->crud->addColumn([
    'name' => 'full_name', // there is no full_name column in the db, it's an accessor 
    'type' => 'text',
    'label' => 'Full Name',
    'searchLogic' => function ($query, $column, $searchTerm) {
        $query->orWhere('first_name', 'like', '%'.$searchTerm.'%');
        $query->orWhere('last_name', 'like', '%'.$searchTerm.'%');
    }
]);
```
### What I expected to happen:
the `searchLogic` applied

### What happened:
`searchLogic` always return `false`

### What I've already tried to fix it:
Modify Columns Traits
```
        // check if the column exists in the DB table
        if (\Schema::hasColumn($this->model->getTable(), $column_with_details['name'])) {
            $column_with_details['tableColumn'] = true;
        } else {
            $column_with_details['tableColumn'] = false;
            $column_with_details['orderable'] = false;
//            $column_with_details['searchLogic'] = false;  // This line always override the closure
        }
```

**Comment by lloy0076** [comment]
This should hit:

https://github.com/Laravel-Backpack/CRUD/blob/bd2badc5ae6a47bb31a62b5af28361d890f4ce63/src/PanelTraits/Search.php#L36

I can't see why it wouldn't.

If you are able, try:

```
            if (is_callable($searchLogic)) {
                dd($searchLogic);
                return $searchLogic($query, $column, $searchTerm);
            }
```

(or get a debugger to work - but `dd` may be much less fussing about).

This will tell you if that is being called.

If it is then try something like:

```
            if (is_callable($searchLogic)) {
                $it = $searchLogic($query, $column, $searchTerm);
                info($it->toSql());
                info($it->getBindings());
                return $it; // or still dd($it);
            }
```

And make sure that the search logic function is getting called with the values you expect. And make sure that the thing you are searching for does actually exist 😄

**Comment by kiddtang** [comment]
I am also wonder why it doesn't work.

```
        if (isset($column['searchLogic'])) {
            $searchLogic = $column['searchLogic'];
            dd($searchLogic);
            if (is_callable($searchLogic)) {
                return $searchLogic($query, $column, $searchTerm);
            }

            if ($searchLogic == false) {
                return;
            }
        }
```
i got `false` on this debugging, it never get in to ` if (is_callable($searchLogic))`

**Comment by kiddtang** [comment]
https://github.com/Laravel-Backpack/CRUD/blob/bd2badc5ae6a47bb31a62b5af28361d890f4ce63/src/PanelTraits/Columns.php#L89-L95

I think the Column Traits need to check user declared the `searchLogic` or not... this is what how i fix the issue...
```
        // check if the column exists in the DB table
        if (\Schema::hasColumn($this->model->getTable(), $column_with_details['name'])) {
            $column_with_details['tableColumn'] = true;
        } else {
            $column_with_details['tableColumn'] = false;
            $column_with_details['orderable'] = false;
            if (! array_key_exists('searchLogic', $column_with_details)) {  // Check before set to false
                $column_with_details['searchLogic'] = false;
            }
        }
```

**Comment by jkudish** [comment]
I have the same issue here

**Comment by indra1** [comment]
@jkudish @kiddtang  I was able to implement it like this:

` 'searchLogic' => function ($q, $column, $searchTerm) {
                    $q->orWhereHas('user', function ($query) use ($searchTerm) {
                        $query->where('first_name', 'like', '%' . $searchTerm . '%');
                        $query->orWhere('last_name', 'like', '%' . $searchTerm . '%');
                    });
                },`

**Comment by jkudish** [comment]
Thank you!

**Comment by matism** [comment]
I'm using the searchLogic as explained in the "Upgrading CRUD 3.2 to 3.3" guide but it is not working for me. I'm only able to fix it by modifying the column traits. What am I missing here?

`$this->crud->addColumn([
    'name' => 'full_name', // there is no full_name column in the db, it's an accessor 
    'type' => 'text',
    'label' => 'Full Name',
    'searchLogic' => function ($query, $column, $searchTerm) {
        $query->orWhere('first_name', 'like', '%'.$searchTerm.'%');
        $query->orWhere('last_name', 'like', '%'.$searchTerm.'%');
    }
]);`

**Comment by indra1** [comment]
@matism  did you use the appends property to add the full_name? If you did it should work

**Comment by matism** [comment]
@indra1 Yes I've added full_name to the appends property. But still `$this->hasColumn($this->model->getTable(), $column_with_details['name'])` returns false.

**Comment by indra1** [comment]
@matism  please open a stack overflow question, add the code and write where you think you have an issue and tag it with backpack-for-laravel 
I just updated my backpack from 3.1 to 3.3 and it works seamlessly

**Comment by kiddtang** [comment]
@matism let's discuss here...
https://stackoverflow.com/questions/47835198/backpack-crud-searchlogic-always-considered-not-declared

@indra1 I met same issue again while this time I need to search on "currency" (a relationship) & "amount" (own table). Hope you could give me some hints on the stackoverflow. Thanks in advance!

**Comment by tabacitu** [comment]
@kiddtang - I think you were totally right on this, I'm sorry I didn't see this earlier. I just [pushed a fix to this](https://github.com/Laravel-Backpack/CRUD/commit/90edeab7f212963807ba140216e10334871f5acc), will be published at the end of the day, together with a few other fixes.

Guys, let me rephrase this the way I understand it:
- when a column is NOT in the database (say... it's mutator; or it's an n-n relationship)
- searchLogic got overwritten by the ```addColumn()``` method;
- so if you defined a searchLogic, it didn't matter, 'cause it got overwritten;

Fixed now. We now place those values ONLY IF they're not explicitly defined. Thanks for bringing this @kiddtang .

Cheers!

**Comment by matism** [comment]
@tabacitu Thanks a lot! Now it's working for me as well.
